Buildings/Experimental/DHC/EnergyTransferStations/Combined/Generation5/BaseClasses/PartialParallel.mo

```diff
@@ -0,0 +1,405 @@
+    annotation (Dialog(group="District heat exchanger"));
+  parameter Modelica.SIunits.Temperature T_b2Hex_nominal
+    "Nominal water outlet temperature on building side"
+    annotation (Dialog(group="District heat exchanger"));
+  parameter Modelica.SIunits.TemperatureDifference dT1HexSet[2]
+    "Primary side deltaT set point schedule (index 1 for heat rejection)"
+    annotation (Dialog(group="District heat exchanger"));
+  parameter Real spePum1HexMin(
+    final unit="1",
+    min=0)=0.1
+    "Heat exchanger primary pump minimum speed (fractional)"
+    annotation (Dialog(group="District heat exchanger",enable=not have_val1Hex));
+  parameter Real yVal1HexMin(
+    final unit="1",
+    min=0.01)=0.1
+    "Minimum valve opening for temperature measurement (fractional)"
+    annotation (Dialog(group="District heat exchanger",enable=have_val1Hex));
+  parameter Real spePum2HexMin(
+    final unit="1",
+    min=0.01)=0.1
+    "Heat exchanger secondary pump minimum speed (fractional)"
+    annotation (Dialog(group="District heat exchanger"));
+  replaceable parameter Buildings.Fluid.Movers.Data.Generic perPum1Hex(
+    motorCooledByFluid=false)
+    constrainedby Buildings.Fluid.Movers.Data.Generic
+    "Record with performance data for primary pump"
+    annotation (Dialog(group="District heat exchanger",enable=not have_val1Hex),
+      choicesAllMatching=true,
+      Placement(transformation(extent={{-80,222},{-60,242}})));
+  replaceable parameter Buildings.Fluid.Movers.Data.Generic perPum2Hex(
+    motorCooledByFluid=false)
+    constrainedby Buildings.Fluid.Movers.Data.Generic
+    "Record with performance data for secondary pump"
+    annotation (Dialog(group="District heat exchanger"),choicesAllMatching=true,
+      Placement(transformation(extent={{-40,222},{-20,242}})));
+  parameter Modelica.SIunits.Volume VTanHeaWat
+    "Heating water tank volume"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Modelica.SIunits.Length hTanHeaWat=
+    (VTanHeaWat*16/Modelica.Constants.pi)^(1/3)
+    "Heating water tank height (assuming twice the diameter)"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Modelica.SIunits.Length dInsTanHeaWat=0.1
+    "Heating water tank insulation thickness"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Modelica.SIunits.Volume VTanChiWat
+    "Chilled water tank volume"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Modelica.SIunits.Length hTanChiWat=
+    (VTanChiWat*16/Modelica.Constants.pi)^(1/3)
+    "Chilled water tank height (without insulation)"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Modelica.SIunits.Length dInsTanChiWat=0.1
+    "Chilled water tank insulation thickness"
+    annotation (Dialog(group="Buffer Tank"));
+  parameter Integer nSegTan=3
+    "Number of volume segments for tanks"
+    annotation (Dialog(group="Buffer Tank"));
+  // IO VARIABLES
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uHea
```

Comment:
may need to run the modelica-formatter to change this?

Reply:
Addressed